### PR TITLE
refactor: action handlers integrated inside actions

### PR DESCRIPTION
**Pull request by matheusribeirozup**
**Description**
Before, every action of Beagle needed a handler to take care of its execution, now the contract has been changed and each one already knows for itself what to perform.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/sample/src/main/java/br/com/zup/beagle/sample/AppCustomActionHandler.kt [hunk 1]
```diff
@@ -17,21 +17,21 @@
...
 import br.com.zup.beagle.action.Navigate
 import br.com.zup.beagle.action.Route
+import br.com.zup.beagle.android.action.FormLocalActionHandler
 import br.com.zup.beagle.android.annotation.BeagleComponent
 import br.com.zup.beagle.widget.layout.NavigationBar
 import br.com.zup.beagle.widget.layout.Screen
 import br.com.zup.beagle.widget.ui.Text
 import java.lang.Thread.sleep
 
 @BeagleComponent
```
Comment (botbeaglezup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/AppCustomActionHandler.kt:31]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/f101225c4445ff422c87672c9d57e5edc78e83ae/android/sample/src/main/java/br/com/zup/beagle/sample/AppCustomActionHandler.kt#L31"></span>The file name ‘AppCustomActionHandler.kt’ does not match the name of the single top-level declaration ‘AppFormLocalActionHandler’.</td>
    </tr>
  </tbody>
</table>

Comment (botbeaglezup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/AppCustomActionHandler.kt:31]: <table data-meta="generated_by_backend">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/f101225c4445ff422c87672c9d57e5edc78e83ae/android/sample/src/main/java/br/com/zup/beagle/sample/AppCustomActionHandler.kt#L31"></span>The file name ‘AppCustomActionHandler.kt’ does not match the name of the single top-level declaration ‘AppFormLocalActionHandler’.</td>
    </tr>
  </tbody>
</table>

File: android/beagle/src/main/java/br/com/zup/beagle/android/action/Action.kt [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 package br.com.zup.beagle.android.action
 
-import br.com.zup.beagle.action.Action
+import br.com.zup.beagle.android.engine.renderer.RootView
 
-interface ActionHandler<T : Action>
\ No newline at end of file
+interface Action : br.com.zup.beagle.action.Action {
+    fun handle(rootView: RootView)
```
Comment (tuliopereirazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/action/Action.kt:22]: To keep the protocol concise, I think it's better to rename the `handle` to `execute`

Comment (matheusribeirozup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/action/Action.kt]: done

File: schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/action/FormRemoteAction.kt [hunk 3]
```diff
@@ -15,54 +15,22 @@
  */
 
 package br.com.zup.beagle.action
+import br.com.zup.beagle.core.ServerDrivenComponent
 
 /**
```
Comment (botbeaglezup) [inline schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/action/FormRemoteAction.kt:20]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/9bcc7993b586e7f5f60449111532229895e3843d/schema/kotlin-core/src/main/kotlin/br/com/zup/beagle/action/FormRemoteAction.kt#L20"></span>The file name ‘FormRemoteAction.kt’ does not match the name of the single top-level declaration ‘Form’.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/9bcc7993b586e7f5f60449111532229895e3843d/android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/ValidationErrorListener.kt#L1">android/beagle/src/main/java/br/com/zup/beagle/android/components/form/core/ValidationErrorListener.kt#L1</a> - The empty Kotlin file ValidationErrorListener.kt can be removed.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**